Scripts/RemediationScripts/Remediate-StopExposingServiceFabricReverseProxyPort.ps1

```diff
@@ -0,0 +1,956 @@
+                                                                          @{N='ReverseProxyPorts';E={if($_.NodeTypes.ReverseProxyEndpointPort.Count -gt 0)
+                                                                          { 
+                                                                                $_.NodeTypes.ReverseProxyEndpointPort
+                                                                          }                                                                          
+                                                                          }}
+            }
+            catch
+            {
+                Write-Host "Error fetching Service Fabric(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }
+        }
+    }
+
+    $totalServiceFabric = ($ServiceFabricDetails| Measure-Object).Count
+
+    if ($totalServiceFabric -eq 0)
+    {
+        Write-Host "No Service Fabric(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+  
+    Write-Host "Found [$($totalServiceFabric)] Service Fabric(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Service Fabric(s) for which Reverse proxy port are exposed.
+    $ServiceFabricWithReverseProxyPortExposed = @()
+
+    Write-Host "Separating Service Fabric(s) for which Reverse proxy port are exposed..."
+    # Collecting all the resource group which are of ERvNet Types.
+    $ERvNetRGNames = @()
+    $ERvNetRGNames += "ERNetwork","ERNetwork-DMZ","ERNetwork-PvtApp","ERNetwork-DB","ERNetwork-InetApp","ERNetwork-SVC","ERNetwork-LAB","ERNetwork-MVD","ERNetwork-WVD"
+
+    $ServiceFabricDetails | ForEach-Object {
+        $ServiceFabric = $_        
+            if($_.IsReverseProxyPortEnabled -eq $true)
+            {
+                # Fetching the Load balancer and rule details 
+                $loadBalancerInCurrentRG = Get-AzLoadBalancer -ResourceGroupName $_.ResourceGroupName
+                $loadBalancerCollection = $loadBalancerInCurrentRG | Where-Object{$_.Tag.Value -eq $_.ResourceName}
+                $isPortExposed = $false
+                $ports = $_.ReverseProxyPorts
+                foreach($lb in $loadBalancerCollection)
+                {
+                    $Rules = $lb.LoadBalancingRules
+                    foreach($element in $ports)
+                    {
+                        if($Rules.BackendPort -contains $element)
+                        {
+                            $isPortExposed = $true
+                        }
+                    }
+                }
+             $ServiceFabricDetails =  Get-AzServiceFabricCluster -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
```

Comment:
$ServiceFabricDetails this variable name is same as the variable which has all the SF. Might create confusion.

Reply:
Changed as mentioned.